consensus/istanbul/core/vrank_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"encoding/hex"
+	"math/big"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/klaytn/klaytn/consensus/istanbul/validator"
+	"github.com/stretchr/testify/assert"
+)
+
+func genCommitteeFromAddrs(addrs []common.Address) istanbul.Validators {
+	committee := []istanbul.Validator{}
+	for _, addr := range addrs {
+		committee = append(committee, validator.New(addr))
+	}
+	return committee
+}
+
+func TestVrank(t *testing.T) {
+	var (
+		N         = 6
+		quorum    = 4
+		addrs, _  = genValidators(N)
+		committee = genCommitteeFromAddrs(addrs)
+		view      = istanbul.View{Sequence: big.NewInt(0), Round: big.NewInt(0)}
+		msg       = &istanbul.Subject{View: &view}
+		vrank     = NewVrank(view, committee)
+
+		expectedAssessList  []int
+		expectedLateCommits []time.Duration
+	)
+
+	sort.Sort(committee)
+	for i := 0; i < quorum; i++ {
+		vrank.AddCommit(msg, committee[i])
+		expectedAssessList = append(expectedAssessList, vrankArrivedEarly)
+	}
+	vrank.HandleCommitted(view.Sequence)
+	for i := quorum; i < N; i++ {
+		t := vrank.AddCommit(msg, committee[i])
+		expectedAssessList = append(expectedAssessList, vrankArrivedLate)
+		expectedLateCommits = append(expectedLateCommits, t)
```

Comment:
What about removing the return type of AddCommit and change the test code like this? It looks better to me.
```
		vrank.AddCommit(msg, committee[i])
		expectedAssessList = append(expectedAssessList, vrankArrivedLate)
		expectedLateCommits = append(expectedLateCommits, vrank.commitArrivalTimeMap[committee[i].Address()])
```

Reply:
Right, I reluctantly added the return type of AddCommit just because of this. updated in 23e6513a